backend/services/durable.py

```
@@ -0,0 +1,86 @@
+_dbos_step = None
+
+if _ENABLED:
+    try:
+        from dbos import DBOS  # type: ignore[import-not-found]
+        # DBOS init must be done by the application entrypoint; we just
+        # capture the decorators here and trust that DBOS() was called
+        # in main.py BEFORE any decorated function is invoked.
+        _dbos_workflow = DBOS.workflow
+        _dbos_step = DBOS.step
+        _HAS_DBOS = True
+    except Exception as e:  # ImportError or DBOS init failure
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)